common/perf-test-core/src/main/java/com/azure/perf/test/core/MockEventProcessor.java

```diff
@@ -0,0 +1,150 @@
+
+        mockEventContexts = new MockEventContext[partitions];
+        IntStream.range(0, partitions).boxed().forEach(integer -> {
+            mockEventContexts[integer] = new MockEventContext(integer, "Hello");
+        });
+        this.eventsRaised = new int[partitions];
+    }
+
+    /**
+     * Starts the event processor.
+     */
+    public synchronized void start() {
+        eventsRaised = new int[eventsRaised.length];
+        process = true;
+        errorRaised = false;
+        startTime = System.nanoTime();
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        scheduler.set(executor);
+
+        Double jitterInMillis =
+            ThreadLocalRandom.current().nextDouble() * TimeUnit.SECONDS.toMillis(0);
+
+        runner.set(scheduler.get().schedule(this::processEvents,
+            jitterInMillis.longValue(), TimeUnit.MILLISECONDS));
+    }
+
+    private Mono<Void> processEvents() {
+        while (process) {
+            for (int i = 0; i < partitions; i++) {
+                process(i);
+            }
+        }
+        return Mono.empty();
+    }
+
+    private void process(int partition) {
+        MockEventContext mockEventContext = mockEventContexts[partition];
+
+        if (maxEventsPerSecond > 0) {
+            while (process) {
+                long elapsedTime = (System.nanoTime() - startTime);
+                if (errorAfter != null && !errorRaised
+                    && (errorAfter.compareTo(Duration.ofNanos(elapsedTime)) < 0)) {
+                    errorLock.lock();
+                    if (!errorRaised) {
+                        processError(partition, new IllegalStateException("Test Exception"));
+                        errorRaised = true;
+                    }
+                }
+                int eventsSent = eventsRaised[partition];
+                double targetEventsSent = ((double) (elapsedTime / 1_000_000_000))
+                    * maxEventsPerSecondPerPartition;
+                if (eventsSent < targetEventsSent) {
+                    processEvent.accept(mockEventContext);
+                    eventsRaised[partition]++;
+                } else {
+                    try {
+                        Thread.sleep((long) ((1 / maxEventsPerSecondPerPartition) * 1000));
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+        } else {
+            while (process) {
+                if (errorAfter != null && !errorRaised
+                    && (errorAfter.compareTo(Duration.ofNanos((System.nanoTime() - startTime))) < 0)) {
+                    errorLock.lock();
+                    if (!errorRaised) {
+                        processError(partition, new IllegalStateException("Test Exception"));
+                        errorRaised = true;
+                    }
+                }
+                processEvent.accept(mockEventContext);
```

Comment:
This should be in an `else` block of `if (errorAfter` above.  See the .NET version:

https://github.com/Azure/azure-sdk-for-net/blob/main/common/Perf/Azure.Sample.Perf/Event/MockEventProcessor.cs#L131